compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -327,8 +326,7 @@ impl<'ll> CodegenCx<'ll, '_> {
                 // ThinLTO can't handle this workaround in all cases, so we don't
                 // emit the attrs. Instead we make them unnecessary by disallowing
                 // dynamic linking when linker plugin based LTO is enabled.
-                !self.tcx.sess.opts.cg.linker_plugin_lto.enabled() &&
-                self.tcx.sess.lto() != Lto::Thin;
```

Comment:
Given the history of this, is it worth explaining that we explicitly do *not* want to check `sess.lto()` here?

Reply:
I rewrote this comment and mentioned that.